tests/src/test/scala/whisk/core/containerpool/docker/test/DockerContainerTests.scala

```diff
@@ -0,0 +1,640 @@
+import java.io.IOException
+
+/**
+ * Unit tests for ContainerPool schedule
+ */
+@RunWith(classOf[JUnitRunner])
+class DockerContainerTests extends FlatSpec
+    with Matchers
+    with MockFactory
+    with StreamLogging
+    with ScalaFutures
+    with BeforeAndAfterEach {
+
+    override def beforeEach() = {
+        stream.reset()
+    }
+
+    /** Awaits the given future, throws the exception enclosed in Failure. */
+    def await[A](f: Future[A], timeout: FiniteDuration = 100.millisecond) = Await.result[A](f, timeout)
+
+    val containerId = ContainerId("id")
+    /**
+     * Constructs a testcontainer with overridden IO methods. Results of the override can be provided
+     * as parameters.
+     */
+    def dockerContainer(
+        id: ContainerId = containerId,
+        ip: ContainerIp = ContainerIp("ip"))(
+            ccRes: Future[RunResult] = Future.successful(RunResult(intervalOf(1.millisecond), Right(ContainerResponse(true, "", None)))),
+            retryCount: Int = 0)(
+                implicit docker: DockerApiWithFileAccess, runc: RuncApi): DockerContainer = {
+
+        new DockerContainer(id, ip) {
+            override protected def callContainer(path: String, body: JsObject, timeout: FiniteDuration, retry: Boolean = false): Future[RunResult] = {
+                ccRes
+            }
+            override protected val logsRetryCount = retryCount
+            override protected val logsRetryWait = 0.milliseconds
+        }
+    }
+
+    /** Creates an interval starting at EPOCH with the given duration. */
+    def intervalOf(duration: FiniteDuration) = Interval(Instant.EPOCH, Instant.ofEpochMilli(duration.toMillis))
+
+    behavior of "DockerContainer"
+
+    implicit val transid = TransactionId.testing
+
+    /*
+     * CONTAINER CREATION
+     */
+    it should "create a new instance" in {
+        implicit val docker = new TestDockerClient
+        implicit val runc = stub[RuncApi]
+
+        val image = "image"
+        val memory = 128.MB
+        val cpuShares = 1
+        val environment = Map("test" -> "hi")
+        val network = "testwork"
+        val name = "myContainer"
+
+        val container = DockerContainer.create(
+            transid = transid,
+            image = image,
+            memory = memory,
+            cpuShares = cpuShares,
+            environment = environment,
+            network = network,
+            name = Some(name))
+
+        await(container)
+
+        docker.pulls should have size 0
+        docker.runs should have size 1
+        docker.inspects should have size 1
+        docker.rms should have size 0
+
+        val (testImage, args) = docker.runs.head
+        testImage shouldBe "image"
+
+        // Assert fixed values are passed as well
+        args should contain allOf ("--cap-drop", "NET_RAW", "NET_ADMIN")
+        args should contain inOrder ("--ulimit", "nofile=1024:1024")
+        args should contain inOrder ("--pids-limit", "1024") // OW PR 2119
```

Comment:
could consolidate all these into a `Seq(x, y).foreach args should contain inOrder (x, y)`